dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKUtilities.java

```diff
@@ -207,4 +220,542 @@ public static TCKJarInfo parseTCKDependencies(Type type, String[] dependencyOutp
+
+            //load the properties file
+
+            try (FileInputStream fis = new FileInputStream(wrapperPropertiesFile)) {
+                props.load(fis);
+            }
+
+            //get the existing value of the distributionUrl property
+            String distributionURL = props.getProperty(MVN_DISTRIBUTION_URL_KEY);
+            //substitute the server string
+            assertThat("distributionURL update failed", distributionURL, containsString(MVN_WRAPPER_REPO));
+            distributionURL = distributionURL.replace(MVN_WRAPPER_REPO, artifactoryRepoURL);
+            //set it back into the properties
+            props.setProperty(MVN_DISTRIBUTION_URL_KEY, distributionURL);
+            Log.info(c, "updatePropertiesFile", MVN_DISTRIBUTION_URL_KEY + "=" + distributionURL);
+
+            //get the existing value of the wrapperUrl property
+            String wrapperURL = props.getProperty(MVN_WRAPPER_URL_KEY);
+            //substitute the server string
+            assertThat("wrapperURL update failed", wrapperURL, containsString(MVN_WRAPPER_REPO));
+            wrapperURL = wrapperURL.replace(MVN_WRAPPER_REPO, artifactoryRepoURL);
+            //set it back into the properties
+            props.setProperty(MVN_WRAPPER_URL_KEY, wrapperURL);
+            Log.info(c, "updatePropertiesFile", MVN_WRAPPER_URL_KEY + "=" + wrapperURL);
+
+            //write the properties back out into the original file
+            try (FileOutputStream fos = new FileOutputStream(wrapperPropertiesFile)) {
+                props.store(fos, "MVN Wrapper Properties");
+            }
+        }
+        return props;
+    }
+
+    public static boolean useArtifactory() {
+        Log.info(c, "useArtifactory", "Force external: " + System.getProperty(FAT_TEST_PREFIX + ARTIFACTORY_FORCE_EXTERNAL_KEY, ""));
+        boolean forceExternal = Boolean.getBoolean(FAT_TEST_PREFIX + ARTIFACTORY_FORCE_EXTERNAL_KEY);
+        if (forceExternal) {
+            return false;
+        }
+
+        return !System.getProperty(FAT_TEST_PREFIX + ARTIFACTORY_SERVER_KEY, "").isEmpty();
+    }
+
+    public static String getArtifactoryServer() {
+        return System.getProperty(FAT_TEST_PREFIX + ARTIFACTORY_SERVER_KEY);
+    }
+
+    public static String getArtifactoryUser() {
+        return System.getProperty(FAT_TEST_PREFIX + ARTIFACTORY_USER_KEY);
+    }
+
+    public static String getArtifactoryToken() {
+        return System.getProperty(FAT_TEST_PREFIX + ARTIFACTORY_TOKEN_KEY);
+    }
+
+    /**
+     * Get the location within the dev directory to create a temporary maven home directory.
+     *
+     * @return             {@code .m2} within the dev directory
+     * @throws IOException
+     */
+    public static File getTemporaryMavenHomeDir() throws IOException {
+        // user.dir is set to the test project directory when running a FAT
+        String userDir = System.getProperty("user.dir");
```

Comment:
consider using a different starting point

Reply:
On re-review I think this is actually correct, `user.dir` is the current working directory.

However, `File` will automatically resolve against this directory when an absolute file path is needed so I'm removing the manual system property lookup and adding some validation of the final directory.